packages/clerk-js/src/utils/__tests__/url.test.ts

```diff
@@ -431,7 +431,22 @@ fdescribe('isAllowedRedirectOrigin', () => {
     ['https://test.clerk.com/foo?hello=1', [/https:\/\/www\.clerk\.com/], false],
   ];
 
+  const warnMock = jest.spyOn(global.console, 'warn').mockImplementation();
+
+  beforeEach(() => warnMock.mockClear());
+  afterAll(() => warnMock.mockRestore());
+
   test.each(cases)('isAllowedRedirectOrigin("%s","%s") === %s', (url, allowedOrigins, expected) => {
     expect(isAllowedRedirectOrigin(url, allowedOrigins)).toEqual(expected);
+
+    if (!expected) {
+      const output = warnMock.mock.calls[0][0];
+
+      expect(warnMock).toHaveBeenCalledTimes(1);
+      expect(output.includes(new URL(url))).toBe(true);
+      expect(output).toMatch(/allowedRedirectOrigins/);
+    } else {
+      expect(warnMock).not.toHaveBeenCalled();
+    }
```

Comment:
🙃 i would avoid having `if/else` in tests since they become more difficult to read and create a possibility of the actual usecase to not be called.
I think in the case of `console.warn`, there is no need to test that it's being called with the appropriate message.
We could refactor the above assertions to 
```typescript
    // ...
    expect(isAllowedRedirectOrigin(url, allowedOrigins)).toEqual(expected);
    expect(warnMock).toHaveBeenCalledTimes(Number(expected)); // Number(true) -> 1 , Number(false) -> 0
 ```
 
 cc: @nikosdouvlis